src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -24,18 +24,32 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
+
+/**
+ * This is the main entry point of the monitor.  It reads the configuration and manages the life cycle of the monitoring
+ * applications.
+ */
 public class KafkaMonitor {
   private static final Logger LOG = LoggerFactory.getLogger(KafkaMonitor.class);
   public static final String CLASS_NAME_CONFIG = "class.name";
 
   private final Map<String, App> _tests;
   private final Map<String, Service> _services;
   private final ScheduledExecutorService _executor;
+  /** When true this instance of the monitor is shutdown and can not be restarted. */
+  private boolean _isShutdown = false;
 
-  public KafkaMonitor(Map<String, Map> testProps) throws Exception {
+  public KafkaMonitor() {
     _tests = new HashMap<>();
     _services = new HashMap<>();
+    _executor = Executors.newSingleThreadScheduledExecutor();
+  }
 
+  public synchronized void configure(Map<String, Map> testProps) throws Exception {
```

Comment:
Can you explain the reason of having `configure(...)`? Do we expect this method to be called multiple times?